DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.WebAPI/Services/GitHubService.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using GitHubExtension.Activity.External.WebAPI.Exceptions;
+using GitHubExtension.Security.WebApi.Library.Exceptions;
+using Newtonsoft.Json.Linq;
+
+namespace GitHubExtension.Activity.External.WebAPI.Services
+{
+    public class GitHubService : IGitHubService
+    {
+        private const string LastLinkRelRegex = "last";
+        private const string LinklRelPageRegex = @"page=(\d+)";
+        private readonly HttpClient _httpClient;
+        private IEnumerable<string> _lastRequestLinkHeader;
+ 
+        private static readonly Dictionary<string, string> DefaultHeaders = new Dictionary<string, string>()
+        {
+             //Need to set user-agent to access GitHub API, Using Chrome 48
+            { "User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.116 Safari/537.36" }
+        };
+
+        private const string EventsUrl = "https://api.github.com/repos/{0}/events?access_token={1}&page={2}";
+
+        public GitHubService()
+        {
+            _httpClient = new HttpClient();
```

Comment:
I also thought that it is Disposable but according to [this ](http://stackoverflow.com/questions/15705092/do-httpclient-and-httpclienthandler-have-to-be-disposed) it's not